### PR TITLE
Add strain rate operator

### PR DESCRIPTION
**Pull request by timofeymukha**

### CODE REVIEW

File: src/math/operators.f90 [hunk 1]
```diff
@@ -221,4 +221,58 @@ function cfl(dt, u, v, w, Xh, coef, nelv, gdim)
...
+    integer :: i, e
+    
+    ! we use s11 as a work array here
+    call dudxyz (s12, u%x, coef%drdy, coef%dsdy, coef%dtdy, coef)
+    call dudxyz (s11, v%x, coef%drdx, coef%dsdx, coef%dtdx, coef)
+    do e=1,u%msh%nelv
+      do i=1,u%Xh%lxyz
+        s12(i, 1, 1, e) = s12(i, 1, 1, e) + s11(i, 1, 1, e) ! du/dy + dv/dx
+       enddo
+    enddo
```
Comment (njansson) [inline src/math/operators.f90:246]: Shouldn't this have if else for gpu calls?

File: src/math/operators.f90 [hunk 2]
```diff
@@ -221,4 +221,58 @@ function cfl(dt, u, v, w, Xh, coef, nelv, gdim)
...
+
+    call dudxyz (s23, v%x, coef%drdz, coef%dsdz, coef%dtdz, coef)
+    call dudxyz (s11, w%x, coef%drdy, coef%dsdy, coef%dtdy, coef)
+    do e=1,u%msh%nelv
+      do i=1,u%Xh%lxyz
+        s23(i, 1, 1, e) = s23(i, 1, 1, e) + s11(i, 1, 1, e) ! dv/dz + dw/dy
+       enddo
+    enddo
```
Comment (njansson) [inline src/math/operators.f90:262]: same

File: src/math/operators.f90 [hunk 3]
```diff
@@ -221,4 +221,58 @@ function cfl(dt, u, v, w, Xh, coef, nelv, gdim)
...
+    call dudxyz (s22, v%x, coef%drdy, coef%dsdy, coef%dtdy, coef)
+    call dudxyz (s33, w%x, coef%drdz, coef%dsdz, coef%dtdz, coef)
+
+    do e=1,u%msh%nelv
+      do i=1,u%Xh%lxyz
+        s11(i, 1, 1, e) = 2 * s11(i, 1, 1, e) ! 2* du/dx
+        s22(i, 1, 1, e) = 2 * s22(i, 1, 1, e) ! 2* dv/dy
+        s33(i, 1, 1, e) = 2 * s33(i, 1, 1, e) ! 2* dv/dy
+       enddo
+    enddo
```
Comment (njansson) [inline src/math/operators.f90:274]: gpu?

**Review by njansson — CHANGES_REQUESTED** [review]
How about launching a larger kernel that calls dudxyz + the assignment instead of doing that on the cpu?

**Comment by timofeymukha** [comment]
> How about launching a larger kernel that calls dudxyz + the assignment instead of doing that on the cpu?

Just so this now :-P.

**Review by njansson — APPROVED** [review]

**Pull request merged**